tests/kernel.rs

```diff
@@ -102,3 +120,473 @@ fn kernel() {
 
     assert!(res.data.is_none());
 }
+
+#[test]
+fn test_fixed_amount_splitter_local_with_message_and_funds() {
```

Comment:
There's a pretty substantial amount of redundancy in these tests. Can we either look at table driven tests or try and refactor these to move replicated code to separate methods?